plugins/dialog/src/models.rs

```diff
@@ -61,8 +61,49 @@ pub enum MessageDialogButtons {
     OkCancel,
     /// 2 buttons `Yes` and `No` with OS default dialog texts
     YesNo,
+    /// 3 buttons `Yes`, `No` and `Cancel` with OS default dialog texts
+    YesNoCancel,
     /// A single `Ok` button with custom text
     OkCustom(String),
     /// 2 buttons `Ok` and `Cancel` with custom texts
     OkCancelCustom(String, String),
+    /// 3 buttons `Yes`, `No` and `Cancel` with custom texts
+    YesNoCancelCustom(String, String, String),
+}
+
+/// Result of a message dialog
+#[derive(Debug, Default, Clone, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum MessageDialogResult {
+    Yes,
+    No,
+    Ok,
+    #[default]
+    Cancel,
+    Custom(String),
+}
```

Comment:
The untagged needs to be on the field or other fields will become `null`

```suggestion
#[derive(Debug, Default, Clone, PartialEq, Eq, Deserialize, Serialize)]
pub enum MessageDialogResult {
    Yes,
    No,
    Ok,
    #[default]
    Cancel,
    #[serde(untagged)]
    Custom(String),
}
```